packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -40,6 +43,17 @@ export function createAuthMiddleware({
       ) {
         ctx.throw(403, 'Insufficient Grant')
       }
+      const grantRef = await grantReferenceService.get(grant.grant)
+      if (grantRef) {
+        if (grantRef.clientId !== grant.clientId) {
+          ctx.throw(409, 'Unknown Client ID')
```

Comment:
Should this instead log and throw `500`?
It seems like more an internal conflict with the AS than the client

Reply:
I was on the fence between the two.